src/Microsoft.Azure.ServiceBus/MessageReceivePump.cs

```diff
@@ -248,13 +254,7 @@ async Task RenewMessageLockTask(Message message, CancellationToken renewLockCanc
                 {
                     MessagingEventSource.Log.MessageReceiverPumpRenewMessageException(this.messageReceiver.ClientId, message, exception);
 
-                    // TaskCanceled is expected here as renewTasks will be cancelled after the Complete call is made.
-                    // ObjectDisposedException should only happen here because the CancellationToken was disposed at which point 
-                    // this renew exception is not relevant anymore. Lets not bother user with this exception.
-                    if (!(exception is TaskCanceledException) && !(exception is ObjectDisposedException))
```

Comment:
```
(exception is ObjectDisposedException)
```
What happens when you pass an already disposed cancellation token to `Task.Delay()`.?

Reply:
The following will cast ```System.Threading.Tasks.TaskCanceledException: A task was canceled.```.

```cs
using System;
using System.Threading;
using System.Threading.Tasks;

namespace canceledtoken
{
    class Program
    {
        static void Main(string[] args)
        {
            MainAsync(args).GetAwaiter().GetResult();
        }

        static async Task MainAsync(string[] args)
        {
            var cts = new CancellationTokenSource();
            cts.Cancel();
            await Task.Delay(TimeSpan.FromSeconds(1), cts.Token);
        }
    }
}
```

But the next program will ouput ```True```:

```cs
using System;
using System.Threading;
using System.Threading.Tasks;

namespace canceledtoken
{
    class Program
    {
        static void Main(string[] args)
        {
            MainAsync(args).GetAwaiter().GetResult();
        }

        static async Task MainAsync(string[] args)
        {
            var cts = new CancellationTokenSource();
            cts.Cancel();
            var delayTask = await Task
                .Delay(TimeSpan.FromSeconds(1), cts.Token)
                .ContinueWith(t => t, CancellationToken.None);
            Console.WriteLine(delayTask.IsCanceled);
        }
    }
}
```

Reply:
Not sure if I follow, but this feels off. Intentionally passing a disposed object will cause Mayhem. In a real-world scenario, would you expect a disposed token to be passed in?

Reply:
Yes. This can happen as it is a multi threaded scenario and some thread can dispose the token without knowing who else is using it.

@havarnov - Since dispose happens only during closing of receiver, we should be okay with that task throwing `DisposedException` and catching it. So can you add that try-catch back?
We are still solving the hot-path code where every message that is being processed doesn't end up with a throw-catch issue.

Reply:
@nemakam - That's seems like a good trade off. Should probably only catch the DisposedExceptions?